### ISSUE TITLE
.pl-s description and usage don't match

### ISSUE THREAD

**Issue by auscompgeek** [StylishThemes/GitHub-Dark#256]
The syntax highlighting themes describe `.pl-s` as `support`. However, from what I can see, it should actually be `string`. This has caused strings to be highlighted as the wrong colour in rendered code views; for example, the Tomorrow Night themes highlight strings as purple rather than green.

**Comment by silverwind** [comment]
Got a example?

**Comment by auscompgeek** [comment]
I guess this one is simple enough.

```python
print('This here is a string.')
```

**Comment by Mottie** [comment]
If you look at [this demo](https://highlightjs.org/static/demo/), it looks like most of the examples are highlighting strings as green. The only time I noticed the string is not green are in the HTTP, JSON and Nginx examples.

If you want to submit a custom theme, I would be happy to include it.

**Comment by auscompgeek** [comment]
This isn't an issue with the themes themselves, rather, it's an issue with selecting elements incorrectly.

**Comment by Mottie** [comment]
GitHub uses a custom syntax highlighter so we don't have control over that... To get the comments for the class names, I had to email them directly and that's how they labeled that selector ([ref](https://github.com/StylishThemes/GitHub-Dark/issues/197#issuecomment-63717143))

I'm not saying we're not going to change anything. I just don't see where the purple strings are showing up and what changes you are requesting. Maybe submit a pull request so we have a better idea?

**Comment by Mottie** [comment]
Hmm, ok I actually changed my theme to Tomorrow Night and I am seeing the purple strings now... odd that the demo is different.

**Comment by auscompgeek** [comment]
Guess GitHub might've changed the classes on us. It wouldn't be the first time. Could we contact them and see what else has changed since then? `.pl-s` is almost definitely `string`. Here's a couple more test cases:

JS:
```javascript
console.log("Hello, World!");
```

bash:
```bash
echo "Say $(echo 'Hello World!')."
```

**Comment by Mottie** [comment]
Yeah, if you look at #197, you'll see they used to have strings named as `.pl-s1`... 

*sigh* I don't have a lot of time to work on this today, but I'll start with updating the `index.html` demo file and go from there.

**Comment by Mottie** [comment]
Another change I noticed while updating the demo is that the class name of the language has changed. It used to add `"highlight-css"` after syntax highlighting was applied, now it throws a "source" into the class name `"highlight-source-css"` - this applies to all highlighted languages.

Also, so far, the following language class names were changed:

* `highlight-text-html-basic` - used to be `highlight-html`
* `highlight-text-html-php` - used to be `highlight-php`
* `highlight-source-c++` - used to be `highlight-cpp`
* `highlight-source-shell` - used to be `highlight-bash`
* `highlight-text-tex` - used to be `highlight-latex`
* `highlight-source-fortran-modern` - used to be `highlight-fortran`

**Comment by auscompgeek** [comment]
Sounds like the `highlight-` class names are now based on the scope names listed in [grammars.yml](https://github.com/github/linguist/blob/master/grammars.yml) in github/linguist.

**Comment by auscompgeek** [comment]
Hmm. I found a case where `.pl-s1` is used now. Shell sessions:

```console
$ echo 'Hello, World!'
Hello, World!
```

**Comment by auscompgeek** [comment]
Guess it is. Here's some more `.pl-s1`, this time Literate Haskell:

```lhs
Test 1:

\begin{code}
module Main where
main = putStrLn "Hello, World!"
\end{code}

Test 2:

> module Main where
>
> main = putStrLn "Hello, World!"
```

`.pl-s1` isn't styled by GitHub's CSS at all, from what I can tell.

**Comment by auscompgeek** [comment]
FYI, [here](//github.com/github/linguist/issues/1822#issuecomment-66510457)'s another old reference mapping of scopes to CSS class names.

**Comment by Mottie** [comment]
Thanks for the update... I'll try to find some time this weekend to work on a fix.

**Comment by auscompgeek** [comment]
I found [sindresorhus/github-markdown-css](/sindresorhus/github-markdown-css) today. The commit history might be interesting.

**Comment by Mottie** [comment]
Hmm, well after looking at everything, the only thing I can come up with is to just set `.pl-s` to have the same color as `.pl-s1`. Any objections?

**Comment by silverwind** [comment]
I wouldn't do that. `pl-s` is blue in GH's sheet, while `pl-s1` is uncolored (renders as `#333`). Maybe we should just make `pl-s1` the same color as the theme's body color.

**Comment by auscompgeek** [comment]
I concur with @silverwind. However, if we do this, we should also be aware of the fact that shell session output won't be easily distinguishable like it is with GH's CSS.

There *is* one place where `pl-s1` needs to be specifically styled, however – in template strings, a la Ruby:
```ruby
puts "#{x} #{y.infinite ? 'TOO LARGE' : y}"
```

**Comment by auscompgeek** [comment]
Looks like sindresorhus/github-markdown-css@1f877b65224704e6ac5fed1d94194f50e3936f42 is when the `.pl-s` and `.pl-s1` change happened, so I guess it's been in this state since at least April. Scary.

**Comment by auscompgeek** [comment]
At this point in time, I would suggest changing `.pl-s1` to `.pl-s`, and `.pl-s2` to `.pl-s1`.

**Comment by Mottie** [comment]
I just updated the [`github/_template.css`](https://github.com/StylishThemes/GitHub-Dark/commit/4200c59145cc71510e1f34b102cb9724e6c1f4f2) file with the comments (element names) I found in the [`github-dark.css`](https://github.com/primer/github-syntax-dark/blob/master/lib/github-dark.css) file created by GitHub. The new comments were added above the definition, so any in-line comments may be attached to unnecessary definitions.

It does not include a `.pl-s2` class at all. In fact, it seems that the `github-dark.css` file is missing a bunch of definitions that we have included in the template. I added the GitHub-dark theme to the [demo](https://stylishthemes.github.io/GitHub-Dark/) and it *seems* to work fine without those extra definitions.

Maybe, after I get back from my vacation, we can clean up the syntax themes by removing those extra definitions.

**Comment by the-j0k3r** [comment]
Thats some long vacation :)

**Comment by Mottie** [comment]
Perpetual vacation.

**Comment by the-j0k3r** [comment]
Thats called death :)

**Comment by the-j0k3r** [comment]
Is this something we can fix or what?

**Comment by Mottie** [comment]
<img title=":_tuzki_dunno:" src="https://cloud.githubusercontent.com/assets/136959/13203562/02e8f4a2-d881-11e5-8b3a-dd691fb0b52d.gif">

**Comment by auscompgeek** [comment]
I think the sensible thing to do forward would be to use the primer syntax theme output linked above (which, side note, was updated to include a carriage-return scope) as a template for all the syntax themes.

Should also go through and remove rules for scopes that have been removed too.

**Comment by the-j0k3r** [comment]
Ive updated the Github syntax theme a while back now, its needing a cleanup fro unused values, help welcome.

In fact All styles need some cleanup and consistency fixes.
